src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php

```diff
@@ -49,7 +49,19 @@ protected function createController($controller)
             $controller = $this->parser->parse($controller);
         }
 
-        return parent::createController($controller);
+        $resolvedController = parent::createController($controller);
+
+        if (1 === substr_count($controller, ':') && is_array($resolvedController)) {
```

Comment:
hmm indeed. I missed the fact that injection was done in this class in a separate method